src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBaker.java

```diff
@@ -389,10 +304,89 @@ private void checkFurnaces()
         }
     }
 
+    /**
+     * Return the map of furnaces assigned to this hut and the product in it.
+     *
+     * @return a hashmap with BlockPos, BakingProduct.
+     */
+    public Map<BlockPos, BakingProduct> getFurnacesWithProduct()
+    {
+        return Collections.unmodifiableMap(furnaces);
+    }
+
+    /**
+     * Remove a furnace from the building.
+     *
+     * @param pos the position of it.
+     */
+    public void removeFromFurnaces(final BlockPos pos)
+    {
+        furnaces.remove(pos);
+    }
+
+    /**
+     * Clear the furnaces list.
+     */
+    public void clearFurnaces()
+    {
+        furnaces.clear();
+    }
+
+    /**
+     * Remove a product from the furnace.
+     *
+     * @param pos the position the furnace is at.
+     */
+    public void removeProductFromFurnace(final BlockPos pos)
+    {
+        furnaces.replace(pos, null);
+    }
+
+    /**
+     * Return a list of furnaces assigned to this hut.
+     *
+     * @return copy of the list
+     */
+    public List<BlockPos> getFurnaces()
+    {
+        return new ArrayList<>(furnaces.keySet());
+    }
+
+    /**
+     * Get the map of current tasks in the baker.
+     *
+     * @return the map of states and products.
+     */
+    public Map<ProductState, List<BakingProduct>> getTasks()
+    {
+        return Collections.unmodifiableMap(new HashMap<>(tasks));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert this Map to an EnumMap. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1640)